test/track.cpp

```diff
@@ -140,7 +149,37 @@ void test_track() {
 	if (!at2 || !at2->isOpen() || !t1->isOpen())
 		throw runtime_error("Renegotiated track is not open");
 
-	// TODO: Test sending RTP packets in track
+	RtpHeader rtp;
+	rtp._first = 0;
+	rtp.preparePacket();
+	rtp.setPayloadType(96);
+	rtp.setSeqNumber(1);
+	rtp.setTimestamp(3000);
+	rtp.setSsrc(2468);
+
+	if (!t1->send(reinterpret_cast<const std::byte *>(&rtp), sizeof(rtp))) {
+		throw runtime_error("Couldn't send RTP packet");
+	}
+
+	// wait for an RTP packet to be received
+	int count = 0;
+	while (count != 4 && receivedRtpRaw.empty() && attempts--)
+		this_thread::sleep_for(1s);
+
+	if (receivedRtpRaw.empty()) {
+		throw runtime_error("Didn't receive RTP packet on pc2");
+	}
+
+	RtpHeader *receivedRtp = reinterpret_cast<RtpHeader *>(receivedRtpRaw.data());
+
+	if (receivedRtp->version() != rtp.version() || receivedRtp->padding() != rtp.padding() ||
+	    receivedRtp->extension() != rtp.extension() ||
+	    receivedRtp->csrcCount() != rtp.csrcCount() || receivedRtp->marker() != rtp.marker() ||
+	    receivedRtp->payloadType() != rtp.payloadType() ||
+	    receivedRtp->seqNumber() != rtp.seqNumber() ||
+	    receivedRtp->timestamp() != rtp.timestamp() || receivedRtp->ssrc() != rtp.ssrc()) {
+		throw runtime_error("Received RTP packet is different than the packet that was sent");
+	}
```

Comment:
That's pretty ugly and does not check all fields. Tried something like this but it didn't work

```cpp
	if (sizeof(*receivedRtp) != sizeof(rtp) || !memcmp(receivedRtp, &rtp, sizeof(*receivedRtp)))	{ 	
		rtp.log(); 	
		receivedRtp->log(); 	
		throw runtime_error("Received RTP packet is	different than the packet that was sent");
	}
```

Reply:
I followed rtppacketizer.cpp and represent both sent and received RTP packets as vectors so we should be able to safely compare them via memcmp now